test/integration/targets/jinja2_native_types/test_preserving_quotes.yml

```diff
@@ -0,0 +1,14 @@
+- hosts: localhost
+  gather_facts: false
+  tasks:
+    - assert:
+        that:
+          - quoted_str == '"hello"'
+          - empty_quoted_str == '""'
+      vars:
+        third_nested_lvl: '"hello"'
+        second_nested_lvl: "{{ third_nested_lvl }}"
+        first_nested_lvl: "{{ second_nested_lvl }}"
+        quoted_str: "{{ first_nested_lvl }}"
+        empty_quoted_str: '""'
+        empty_str: "{{ empty_quoted_str}}"
```

Comment:
@mkrizek `empty_str` is unused. Did you want to is it for an additional assertion, or in place of `empty_quoted_str == '""'`?